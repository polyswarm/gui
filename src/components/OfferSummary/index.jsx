// Vendor imports
import React, {Component} from 'react';
import PropTypes from 'prop-types';
// Bounty imports
import Button from '../Button';
import StatRow from '../StatRow';
// Component imports
import strings from './strings';

class OfferSummary extends Component {
  constructor(props) {
    super(props);

    this.onPayClick = this.onPayClick.bind(this);
    this.onRequestClick = this.onRequestClick.bind(this);
  }

  render() {
    const {props: {offer}} = this;
    const messages = offer.messages || [];
    const artifacts = messages
      .filter((message) => message.type==='request')
      .map((message) => message.files)
      .reduce((all, files) => all.concat(files), [])
      .sort((a, b) => {
        return a.name.compare(b.name);
      });
    
    messages.filter((message) => message.type==='assertion')
      .forEach((message) => {
        message.files.for_each((file, index) => {
          const verdict = message.verdicts[index];
          const i = artifacts.find((value) => value.hash === file.hash);
          if (i) {
            artifacts[index].verdict = verdict;
          }
        });
      });
    return (
      <div className='OfferSummary'>
        <div className='OfferActions'>
          <Button onClick={this.onPayClick}>
            {strings.pay}
          </Button>
          <Button onClick={this.onRequestClick}>
            {strings.request}
          </Button>
        </div>
        <StatRow vertical
          title={strings.poster}
          content={offer.author}/>
        <StatRow vertical
          title={strings.expert}
          content={offer.expert}/>
        <StatRow vertical
          title={strings.balance}
          content={`${offer.remaining || 0}${strings.nectar}`}/>
        <StatRow vertical
          title={strings.closed}
          content={offer.expired ? strings.yes : strings.no}/>
        <StatRow vertical
          title={strings.messages}
          content={messages.length}/>
        {artifacts.map((artifact) => {
          let content = artifact.verdict ? strings.malicious : strings.safe;
          return(
            <StatRow
              key={artifact.name}
              vertical
              title={artifact.name}
              content={content}/>
          );
        })}
      </div>
    );
  }

  onPayClick() {
    const {props: {onPayClick}} = this;
    if (onPayClick) {
      onPayClick();
    }
  }

  onRequestClick() {
    const {props: {onRequestClick}} = this;
    if (onRequestClick) {
      onRequestClick();
    }
  }
}
OfferSummary.proptypes = {
  offer: PropTypes.object.isRequired,
  onPayClick: PropTypes.func,
  onRequestClick: PropTypes.func,
};
export default OfferSummary;