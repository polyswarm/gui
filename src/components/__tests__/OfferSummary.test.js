import React from 'react';
import {render} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import OfferSummary from '../OfferSummary';

it('renders without crashing', () => {
  const wrapper = render(<OfferSummary />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});