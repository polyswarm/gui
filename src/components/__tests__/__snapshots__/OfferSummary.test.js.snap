// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="OfferSummary"
>
  <li
    class="StatRow-Vertical"
  >
    <p
      class="StatTitle"
    >
      Author:
    </p>
    <p
      class="StatContent"
    >
      author
    </p>
  </li>
  <li
    class="StatRow-Vertical"
  >
    <p
      class="StatTitle"
    >
      Expert:
    </p>
    <p
      class="StatContent"
    >
      expert
    </p>
  </li>
  <li
    class="StatRow-Vertical"
  >
    <p
      class="StatTitle"
    >
      Remaining Balance:
    </p>
    <p
      class="StatContent"
    >
      100 Nectar (NCT)
    </p>
  </li>
  <li
    class="StatRow-Vertical"
  >
    <p
      class="StatTitle"
    >
      
    </p>
    <p
      class="StatContent"
    >
      No
    </p>
  </li>
  <li
    class="StatRow-Vertical"
  >
    <p
      class="StatTitle"
    >
      Messages:
    </p>
    <p
      class="StatContent"
    >
      4
    </p>
  </li>
  <li
    class="StatRow-Vertical"
  >
    <p
      class="StatTitle"
    >
      Benign
    </p>
    <p
      class="StatContent"
    >
      Safe
    </p>
  </li>
  <li
    class="StatRow-Vertical"
  >
    <p
      class="StatTitle"
    >
      Malicious
    </p>
    <p
      class="StatContent"
    >
      Malicious
    </p>
  </li>
</div>
`;
